src/main/java/io/usethesource/vallang/io/binary/message/IValueReader.java

```diff
@@ -349,12 +351,25 @@ private Type readType(final IWireInputStream reader) throws IOException{
                         case IValueIDs.MapType.KEY_TYPE:
                             keyType = readType(reader);
                             break;
+                        case IValueIDs.MapType.KEY_NAME:
```

Comment:
is this not faster if the case keys are ordered? swap KEY_NAME and VALUE_NAME to the third and fourth positions?

Reply:
Compiler does that for us. https://stackoverflow.com/a/23204827/11098